src/test/HostActivationTests/FrameworkResolution/RollForwardMultipleFrameworks.cs

```diff
@@ -0,0 +1,576 @@
+            string rollForward,
+            string resolvedFramework)
+        {
+            RunTest(
+                runtimeConfig => runtimeConfig
+                    .WithFramework(MicrosoftNETCoreApp, "5.1.1")
+                    .WithFramework(MiddleWare, "2.1.0"),
+                dotnetCustomizer => dotnetCustomizer.Framework(MiddleWare).RuntimeConfig(runtimeConfig =>
+                    runtimeConfig.GetFramework(MicrosoftNETCoreApp)
+                        .WithRollForward(rollForward)
+                        .Version = versionReference))
+                .ShouldHaveResolvedFrameworkOrFailedToReconcileFrameworkReference(
+                    MicrosoftNETCoreApp, resolvedFramework, versionReference, "5.1.1");
+        }
+
+        // Verify that inner framework reference (<fxRefVersion>, <rollForward>)
+        // is correctly reconciled with app's framework reference 5.1.1 <defaults>. App fx reference is lower.
+        // Also validates that since all relevant available versions are release, 
+        // the DOTNET_ROLL_FORWARD_TO_PRERELEASE has no effect on the result.
+        [Theory] // fxRefVersion  rollForward                               rollForwadToPreRelease resolvedFramework
+        [InlineData("5.4.0",      null,                                     false,                 "5.4.1")]
+        [InlineData("5.4.0",      Constants.RollForwardSetting.Minor,       false,                 "5.4.1")]
+        [InlineData("5.4.0",      Constants.RollForwardSetting.Minor,       true,                  "5.4.1")]
+        [InlineData("5.4.0",      Constants.RollForwardSetting.LatestMinor, false,                 "5.4.1")] // The app's settings (Minor) wins, so effective reference is "5.4.0 Minor"
+        [InlineData("5.4.0",      Constants.RollForwardSetting.LatestMinor, true,                  "5.4.1")]
+        [InlineData("5.4.0",      Constants.RollForwardSetting.Major,       false,                 "5.4.1")] // The app's settings (Minor) wins, so effective reference is "5.4.0 Minor"
+        [InlineData("5.4.0",      Constants.RollForwardSetting.Major,       true,                  "5.4.1")]
+        [InlineData("5.4.0",      Constants.RollForwardSetting.LatestMajor, false,                 "5.4.1")] // The app's settings (Minor) wins, so effective reference is "5.4.0 Minor"
+        [InlineData("5.4.0",      Constants.RollForwardSetting.LatestMajor, true,                  "5.4.1")]
+        [InlineData("5.4.1",      Constants.RollForwardSetting.Disable,     false,                 "5.4.1")]
+        [InlineData("6.0.0",      Constants.RollForwardSetting.Minor,       false,                 null)]
+        [InlineData("6.0.0",      Constants.RollForwardSetting.Minor,       true,                  null)]
+        [InlineData("6.0.0",      Constants.RollForwardSetting.Major,       false,                 null)]
+        [InlineData("6.0.0",      Constants.RollForwardSetting.LatestMajor, false,                 null)]
```

Comment:
Add cases 
`5.7.0` to `null`
`5.4.1-prerelease` to 5.4.1

Reply:
Added the `5.7.0` cases.
The pre-release to release is covered later on in the `6.1.1-preview.0` kind of tests.